proteus/cmeshToolsModule.cpp

```diff
@@ -1746,15 +1761,14 @@ static PyObject* SparsityInfo_getCSR(SparsityInfo *self,
       for(std::set<int>::iterator sit=self->columnIndecesMap[I].begin();sit != self->columnIndecesMap[I].end();sit++)
         {
           self->columnOffsetsMap[I][*sit] = offset;
-          assert(rowptr_[I]+offset < nnz);
           colind_[rowptr_[I]+offset]=*sit;
           offset++;
         }
       std::sort(&colind_[rowptr_[I]],&colind_[rowptr_[I+1]]);
       max_nonzeros = std::max(max_nonzeros,rowptr_[I+1] - rowptr_[I]);
     }
   //std::cout<<"Proteus: Maximum nonzeros in any row is "<<max_nonzeros<<std::endl;
-  return Py_BuildValue("(O,O,i,O)",PyArray_Return(rowptr),PyArray_Return(colind),nnz,PyArray_Return(nzval));
+  return Py_BuildValue("(N,N,i,N)",PyArray_Return(rowptr),PyArray_Return(colind),nnz,PyArray_Return(nzval));
```

Comment:
That sounds right. Looking at the documentation for `Py_BuildValue`, specifying a "O" increments the reference count on an object on top of the reference created during the object construction, whereas "N" does not. This allows `del` to remove all references to an object to trigger garbage collection